### PR TITLE
Allow other time units in notification and scheduleddowntime

### PR DESCRIPTION
**Pull request by jkroepke**
Hi,

while i'm moving from the old puppet to the new, I have notice that values like 30m are not allowed anymore in some parameters.

Here is a PR for 2 cases, maybe there are a lot of more...

### CODE REVIEW

File: manifests/object/notification.pp [hunk 1]
```diff
@@ -123,7 +123,7 @@
   if !is_string($user_groups) { validate_array($user_groups) }
   if $times { validate_hash ($times )}
   if $command { validate_string ($command )}
-  if $interval { validate_integer ($interval )}
+  if $interval { validate_re($interval, '^\d+\.?\d*[d|h|m|s]?$')}
```
Comment (lazyfrosch) [inline manifests/object/notification.pp]: Please use: `\d+(\.\d+)?[dhms]?`

Your syntax is not really correct, its either `(d|h|m|s)` *or* `[dhms]`

File: manifests/object/scheduleddowntime.pp [hunk 2]
```diff
@@ -88,7 +88,7 @@
   if $author { validate_string($author) }
   if $comment { validate_string($comment) }
   if $fixed { validate_bool($fixed) }
-  if $duration { validate_integer($duration) }
+  if $duration { validate_re($duration, '^\d+\.?\d*[d|h|m|s]?$') }
```
Comment (lazyfrosch) [inline manifests/object/scheduleddowntime.pp]: change the regexp here too

**Review by lazyfrosch — CHANGES_REQUESTED** [review]
Please update the Regexp, but thanks for noticing :+1:

File: manifests/object/notification.pp (hunk 1 above)
Comment (jkroepke) [reply manifests/object/notification.pp]: Okay, I copied it from https://github.com/Icinga/puppet-icinga2/blob/master/manifests/object/service.pp#L216 without thinking,

**Comment by lazyfrosch** [comment]
I guess we should update that everywhere.

Want to do it? Would you have a look on the failing tests, or should we?

**Review by lazyfrosch — APPROVED** [review]

**Comment by jkroepke** [comment]
> Want to do it? Would you have a look on the failing tests, or should we?

Can I look into it.

**Comment by jkroepke** [comment]
@lazyfrosch 

All breakage relevant tests failed, because the error message has changed. I wound why there are test for breakage since you comment long time okay that is usually or unnecessary.

https://github.com/Icinga/puppet-icinga2-legacy/pull/169#issuecomment-177488049

**Comment by lazyfrosch** [comment]
Noticed that myself while activating travis tests. Will discuss that this week with the guys.

Had similar problems with expecting for string behavior between Puppet 3+4.

Could you update the tests so:

* it tests for numeric and with suffix
* does not try to raise and test for exceptions for that params

Appreciate your help here ;)

**Comment by bobapple** [comment]
Hi @jkroepke 
thanks for the effort. We should should rather change the tests where the error message has changed instead of removing them. I don't think these tests are unnecessary, they help us when editing the format of a parameter, like in this case here. I would ask you to please just change the error messages where necessary.

Please let me know if you will look into other objects, if they need this kind of updates. I just want to have a plan how we will proceed with this.

**Comment by bobapple** [comment]
I fixed the specs and merged into master

**Pull request merged**